hack/bats/extras/k8s.bats

```diff
@@ -0,0 +1,64 @@
+# SPDX-FileCopyrightText: Copyright The Lima Authors
+# SPDX-License-Identifier: Apache-2.0
+
+# This test verifies that a Kubernetes cluster can be started and that the single node is ready.
+
+load "../helpers/load"
+
+: "${TEMPLATE:=k8s}"
+
+# Instance names are "${NAME}-0", "${NAME}-1", ...
+NAME="k8s"
+
+get_num_nodes() {
+    local nodes=0
+    for tag in "${BATS_TEST_TAGS[@]}"; do
+        if [[ $tag =~ ^nodes:([0-9]+)$ ]]; then
+            nodes="${BASH_REMATCH[1]}"
+        fi
+    done
+    if [[ $nodes -eq 0 ]]; then
+        echo >&2 "nodes:N tag is required"
+        exit 1
+    fi
+    echo "$nodes"
+}
+
+local_setup() {
+    local nodes=$(get_num_nodes)
+    for ((i=0; i<nodes; i++)); do
+        limactl delete --force "${NAME}-$i" || :
+        limactl start --tty=false --name "${NAME}-$i" "template:${TEMPLATE}" 3>&- 4>&-
+        # NOTE: No support for multi-node clusters yet.
+    done
+    for node in $(k get node -o name); do
+	    k wait --timeout=5m --for=condition=ready "${node}"
+    done
+}
+
+local_teardown() {
+    local nodes=$(get_num_nodes)
+    for ((i=0; i<nodes; i++)); do
+        limactl delete --force "${NAME}-$i" || :
+    done
+}
+
+k() {
```

Comment:
I would call it `kubectl` for readability. Local functions always take precedence over running external programs.